### DISCUSSION TITLE
Bun v0.5.8

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#2436] [Announcements]
**[Read Bun v0.5.8's release notes on Bun's blog](https://bun.sh/blog/bun-v0.5.8)**

To install:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade:
```bash
bun upgrade
```

## What's Changed
* Upgrade Zig by @dylan-conway in https://github.com/oven-sh/bun/pull/2151
* prefer `bun.copy()` over `std.mem.copy()` by @alexlamsl in https://github.com/oven-sh/bun/pull/2152
* [docs] Minor syntax correction in test.md by @jakeboone02 in https://github.com/oven-sh/bun/pull/2153
* fix(zig-build): disable stack probing on x86_64 so no `compiler_rt` needed by @ThatOneBro in https://github.com/oven-sh/bun/pull/2159
* fix(body) Make Request/Reponse empty body to be null by @cirospaciari in https://github.com/oven-sh/bun/pull/2156
* Implement `os.cpus` for Darwin (OS X) by @jwhear in https://github.com/oven-sh/bun/pull/2115
* Fix #2024 by @jwhear in https://github.com/oven-sh/bun/pull/2161
* fix(dns) Fix resolve4 and resolve6 behavior by @cirospaciari in https://github.com/oven-sh/bun/pull/2144
* fix `compiler_rt` linking failure on `x86_64` by @alexlamsl in https://github.com/oven-sh/bun/pull/2163
* Fix typo in `developing.md` by @charliermarsh in https://github.com/oven-sh/bun/pull/2166
* fix(zig-build): disable stack probing when building other objs too by @ThatOneBro in https://github.com/oven-sh/bun/pull/2165
* Add missing "that" in `modules.md` by @charliermarsh in https://github.com/oven-sh/bun/pull/2167
* Docs tweaks by @colinhacks in https://github.com/oven-sh/bun/pull/2160
* Fixes by @colinhacks in https://github.com/oven-sh/bun/pull/2168
* Docs by @colinhacks in https://github.com/oven-sh/bun/pull/2170
* Update hono.md by @ThatOneBro in https://github.com/oven-sh/bun/pull/2171
* Docs: Fixes broken link in express docs by @BrettBlox in https://github.com/oven-sh/bun/pull/2173
* docs: use proper CSS property in React examples by @damianstasik in https://github.com/oven-sh/bun/pull/2177
* docs: Fix sqlite.md `.serialize` typo by @Sheraff in https://github.com/oven-sh/bun/pull/2178
* docs: consistent indentation for post install commands by @johnnyreilly in https://github.com/oven-sh/bun/pull/2175
* docs: fix typo in install.md file by @fdaciuk in https://github.com/oven-sh/bun/pull/2179
* fix: fixes typo in docs/index.md by @akash-joshi in https://github.com/oven-sh/bun/pull/2182
* docs: improved language in yaml loader example by @TommasoAmici in https://github.com/oven-sh/bun/pull/2172
* docs: Fix the link to Node.js ecosystem by @raxityo in https://github.com/oven-sh/bun/pull/2181
* Replace docs/runtime/nodejs with docs/ecosystem/nodejs by @charliermarsh in https://github.com/oven-sh/bun/pull/2185
* fix: Broken link to Node.js support status page by @DreierF in https://github.com/oven-sh/bun/pull/2183
* Fix broken link in developing.md by @rmorey in https://github.com/oven-sh/bun/pull/2188
* improve `bun pm ls` by @alexlamsl in https://github.com/oven-sh/bun/pull/2192
* Update `node:http` compatibility by @ThatOneBro in https://github.com/oven-sh/bun/pull/2184
* docs: fix `globalThis.reloadCount++;` to `globalThis.count++;` by @cunzaizhuyi in https://github.com/oven-sh/bun/pull/2198
* fix ANSI escape codes piped from `stdout` to file by @alexlamsl in https://github.com/oven-sh/bun/pull/2202
* docs(configuration): update misleading comment by @rodoabad in https://github.com/oven-sh/bun/pull/2205
* resolve duplicated dependency correctly by @alexlamsl in https://github.com/oven-sh/bun/pull/2213
* Feat/os.network interfaces by @jwhear in https://github.com/oven-sh/bun/pull/2142
* Bundler uses incorrect port when dev server is listening on non-default port #1885 by @xjmdoo in https://github.com/oven-sh/bun/pull/2191
* Update module statuses in docs by @gaurishhs in https://github.com/oven-sh/bun/pull/2219
* fix(fetch.signal) capture socket and shutdown on abort signal by @cirospaciari in https://github.com/oven-sh/bun/pull/2143
* add signal on http.Server.listen by @cirospaciari in https://github.com/oven-sh/bun/pull/2223
* use abort signal on http.ClientRequest by @cirospaciari in https://github.com/oven-sh/bun/pull/2222
* fix(bun:sqlite): fix `sqliteDb.run(' ')` throwing `not an error` by @ThatOneBro in https://github.com/oven-sh/bun/pull/2226
* Update file-io.md by @maor-benami in https://github.com/oven-sh/bun/pull/2229
* Update file-io.md by @maor-benami in https://github.com/oven-sh/bun/pull/2230
* Implement `preload` support (like `node -r ` except in a config file) by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/2231
* Update clap by @jwhear in https://github.com/oven-sh/bun/pull/2238
* Add `-D`, `--dev` flags for bun install by @jwhear in https://github.com/oven-sh/bun/pull/2240
* fix (globals.d.ts) some comments fix by @cirospaciari in https://github.com/oven-sh/bun/pull/2237
* docs(ecosystem): Update FormData added vers by @ThatOneBro in https://github.com/oven-sh/bun/pull/2236
* Force a specific libdir for c-ares by @jwhear in https://github.com/oven-sh/bun/pull/2241
* fix Bun.file.arrayBuffer() segmentation fault on empty file #2248 by @cirospaciari in https://github.com/oven-sh/bun/pull/2249
* fix deinit behavior when connection is aborted using ResponseStream and abort event behavior by @cirospaciari in https://github.com/oven-sh/bun/pull/2252
* avoids segfault after aborted onReject in Bun.serve streams by @cirospaciari in https://github.com/oven-sh/bun/pull/2256
* fix(node:http/https): fix passing `URL` objs to `http.request`(#2253) by @ThatOneBro in https://github.com/oven-sh/bun/pull/2258
* fix(node:http): match Node `http.request()` GET/HEAD w/ body by @ThatOneBro in https://github.com/oven-sh/bun/pull/2262
* fix bun server segfault with abortsignal by @cirospaciari in https://github.com/oven-sh/bun/pull/2261
* Add page descriptions by @colinhacks in https://github.com/oven-sh/bun/pull/2260
* fix(bindings): mark `JSCInitialize` by @ThatOneBro in https://github.com/oven-sh/bun/pull/2265
* Fix `Bun.sleepSync` to actually use milliseconds by @jwhear in https://github.com/oven-sh/bun/pull/2242
* only free HTTPServerWritable sink after promises solves by @cirospaciari in https://github.com/oven-sh/bun/pull/2268
* place AbortSignal in WebCore namespace like it should by @cirospaciari in https://github.com/oven-sh/bun/pull/2267
* Remake typings for FFI dlopen/linkSymbols + introduce Pointer type by @u9g in https://github.com/oven-sh/bun/pull/2227
* fix tmp dir in fetch.tests.js by @cirospaciari in https://github.com/oven-sh/bun/pull/2270
* add tests for scoped dependency aliasing by @alexlamsl in https://github.com/oven-sh/bun/pull/2271
* Add separate bun-types release CI by @colinhacks in https://github.com/oven-sh/bun/pull/2275
* fix(os.tmpdir()) strip trailing slash on *unix machines by @cirospaciari in https://github.com/oven-sh/bun/pull/2276
* Add `fs/promises` types. Migrate `fs.test.js` to TypeScript. by @colinhacks in https://github.com/oven-sh/bun/pull/2279
* Add a zig fmt action by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/2277
* Fix update zls version in devcontainer by @zhiyuang in https://github.com/oven-sh/bun/pull/2280
* docs: Fix variable name in transpiler API docs by @aabccd021 in https://github.com/oven-sh/bun/pull/2281
* Fix http server req url by @zhiyuang in https://github.com/oven-sh/bun/pull/2285
* Fix broken async test example by @hanneskaeufler in https://github.com/oven-sh/bun/pull/2291
* Improve types for `node:http` by @colinhacks in https://github.com/oven-sh/bun/pull/2284
* fix(bunx): Show usage when calling bunx without arguments by @Zeko369 in https://github.com/oven-sh/bun/pull/2295
* fix(node:http): match Node headers casing (lowercase only) by @ThatOneBro in https://github.com/oven-sh/bun/pull/2288
* fix(bunx): can't resolve executable for scoped packages by @Zeko369 in https://github.com/oven-sh/bun/pull/2293
* [install] use `latest` tagged version if in range by @alexlamsl in https://github.com/oven-sh/bun/pull/2302
* docs(websocket): change server port on example to default port by @aabccd021 in https://github.com/oven-sh/bun/pull/2301
* docs: Fix invalid default port on bun-types docs by @aabccd021 in https://github.com/oven-sh/bun/pull/2300
* Test action comment by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/2310
* [install] fix connection error detection by @alexlamsl in https://github.com/oven-sh/bun/pull/2307
* consider current working directory when resolving relative paths by @alexlamsl in https://github.com/oven-sh/bun/pull/2313
* support `expect().toThrow(/pattern/)` by @alexlamsl in https://github.com/oven-sh/bun/pull/2314
* fix missing `Blob` error messages on Linux by @alexlamsl in https://github.com/oven-sh/bun/pull/2315
* fix & clean up tests by @alexlamsl in https://github.com/oven-sh/bun/pull/2318
* feat(tty): add some `tty.WriteStream` methods to `process.{stdout, stderr}` by @ThatOneBro in https://github.com/oven-sh/bun/pull/2320
* `bun test` on macOS in GitHub Actions by @alexlamsl in https://github.com/oven-sh/bun/pull/2322
* throw invalid parameter errors in `crypto.scryptSync` by @dylan-conway in https://github.com/oven-sh/bun/pull/2331
* Add reference documentation for bun:test by @Electroid in https://github.com/oven-sh/bun/pull/2327
* Reorganize tests by @Electroid in https://github.com/oven-sh/bun/pull/2332
* Fix the --hot example code by @pfgithub in https://github.com/oven-sh/bun/pull/2334
* Improve Benchmarking page, small fixes by @colinhacks in https://github.com/oven-sh/bun/pull/2339
* improve `Buffer` compatibility with Node.js by @alexlamsl in https://github.com/oven-sh/bun/pull/2341
* run `bun test` after macOS builds by @alexlamsl in https://github.com/oven-sh/bun/pull/2343
* More spec compliant `Blob.prototype.type` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/2340
* docs(react): add the missing props.message to `<Component />` by @bushuai in https://github.com/oven-sh/bun/pull/2347
* fix(node:child_process): fix missing primordial `StringPrototypeSlice` by @ThatOneBro in https://github.com/oven-sh/bun/pull/2352
* docs: fix link in globals.md by @Zeko369 in https://github.com/oven-sh/bun/pull/2362
* remove JSON trailing comma in quickstart guide by @noahmarro in https://github.com/oven-sh/bun/pull/2364
* Update file-system-router.md by @harisvsulaiman in https://github.com/oven-sh/bun/pull/2365
* Update comparison matchers compatibility on test doc by @nskins in https://github.com/oven-sh/bun/pull/2366
* fix `require.resolve` with an empty options object #2370 by @paperdave in https://github.com/oven-sh/bun/pull/2371
* docs: Add install with proto example. by @milesj in https://github.com/oven-sh/bun/pull/2361
* feat(net.createServer) and adds socket.connect IPC support by @cirospaciari in https://github.com/oven-sh/bun/pull/2337
* Markdown by @jsoref in https://github.com/oven-sh/bun/pull/2383
* feat(expect): add toBeInstanceOf matcher by @zhiyuang in https://github.com/oven-sh/bun/pull/2389
* Implement `toMatchSnapshot()` by @dylan-conway in https://github.com/oven-sh/bun/pull/2294
* [install] sanitise against malformed `bun.lockb` by @alexlamsl in https://github.com/oven-sh/bun/pull/2397
* allows Bun.serve to listen on IPV6 by @cirospaciari in https://github.com/oven-sh/bun/pull/2400
* add tsconfig to resolve tests and update paths by @dylan-conway in https://github.com/oven-sh/bun/pull/2401
* fix gc-related flaky test failures by @alexlamsl in https://github.com/oven-sh/bun/pull/2402
* feat(expect): update toBeInstanceOf by @zhiyuang in https://github.com/oven-sh/bun/pull/2396
* Fix socket tests with connection errors by @dylan-conway in https://github.com/oven-sh/bun/pull/2403
* Feat(test): add toMatch by @zhiyuang in https://github.com/oven-sh/bun/pull/2404
* Fix various fetch/response/request tests by @dylan-conway in https://github.com/oven-sh/bun/pull/2416
* Fix several bugs by @dylan-conway in https://github.com/oven-sh/bun/pull/2418
* docs: Use correct url in the 'Issues' link in README header by @kamilogorek in https://github.com/oven-sh/bun/pull/2420
* Several bug fixes by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/2427

## New Contributors
* @jakeboone02 made their first contribution in https://github.com/oven-sh/bun/pull/2153
* @charliermarsh made their first contribution in https://github.com/oven-sh/bun/pull/2166
* @BrettBlox made their first contribution in https://github.com/oven-sh/bun/pull/2173
* @damianstasik made their first contribution in https://github.com/oven-sh/bun/pull/2177
* @Sheraff made their first contribution in https://github.com/oven-sh/bun/pull/2178
* @johnnyreilly made their first contribution in https://github.com/oven-sh/bun/pull/2175
* @fdaciuk made their first contribution in https://github.com/oven-sh/bun/pull/2179
* @akash-joshi made their first contribution in https://github.com/oven-sh/bun/pull/2182
* @TommasoAmici made their first contribution in https://github.com/oven-sh/bun/pull/2172
* @raxityo made their first contribution in https://github.com/oven-sh/bun/pull/2181
* @DreierF made their first contribution in https://github.com/oven-sh/bun/pull/2183
* @rmorey made their first contribution in https://github.com/oven-sh/bun/pull/2188
* @cunzaizhuyi made their first contribution in https://github.com/oven-sh/bun/pull/2198
* @rodoabad made their first contribution in https://github.com/oven-sh/bun/pull/2205
* @xjmdoo made their first contribution in https://github.com/oven-sh/bun/pull/2191
* @maor-benami made their first contribution in https://github.com/oven-sh/bun/pull/2229
* @aabccd021 made their first contribution in https://github.com/oven-sh/bun/pull/2281
* @hanneskaeufler made their first contribution in https://github.com/oven-sh/bun/pull/2291
* @Zeko369 made their first contribution in https://github.com/oven-sh/bun/pull/2295
* @pfgithub made their first contribution in https://github.com/oven-sh/bun/pull/2334
* @bushuai made their first contribution in https://github.com/oven-sh/bun/pull/2347
* @noahmarro made their first contribution in https://github.com/oven-sh/bun/pull/2364
* @harisvsulaiman made their first contribution in https://github.com/oven-sh/bun/pull/2365
* @nskins made their first contribution in https://github.com/oven-sh/bun/pull/2366
* @milesj made their first contribution in https://github.com/oven-sh/bun/pull/2361
* @jsoref made their first contribution in https://github.com/oven-sh/bun/pull/2383
* @kamilogorek made their first contribution in https://github.com/oven-sh/bun/pull/2420

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v0.5.7...bun-v0.5.8

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v0.5.8'>Bun v0.5.8</a>.</em>

**Comment by EvHaus** [comment]
Trying to install 0.5.8 on my Mac is giving me:
> bun v0.5.8 is out, but not for this platform (darwin-aarch64) yet.

First time I've seen that error. Is this expected, or did something go wrong with the build?

**Reply by Bellisario** [reply]
Same problem for me. The strage fact is that I'm already on `v0.5.8`...

<img width="697" alt="bug" src="https://user-images.githubusercontent.com/72039923/226987695-1f6d4911-f007-40e7-970d-401fc090fef5.png">

**Reply by jsoref** [reply]
That's because https://github.com/oven-sh/bun/blob/2bdaa81b1c2325687c5115b4e97627533cb3646b/src/cli/upgrade_command.zig#LL194C36-L194C83 is looking for it in https://github.com/Jarred-Sumner/bun-releases-for-updater/releases/tag/bun-v0.5.8 @Jarred-Sumner

**Reply by jsoref** [reply]
Notably https://github.com/Jarred-Sumner/bun-releases-for-updater/releases/tag/bun-v0.5.7 has the darwin-aarch64 file...

**Reply by Jarred-Sumner** [reply]
Fixed

Sorry about this
